tests/waku_rln_relay/test_waku_rln_relay.nim

```diff
@@ -886,3 +886,48 @@ suite "Waku rln relay":
     check:
       buckets.len == 5
       buckets == [2.0, 4.0, 6.0, 8.0, 10.0]
+  
+  asyncTest "nullifierLog clearing only after epoch has passed":
+    let index = MembershipIndex(0)
+
+    proc runTestForEpochSizeSec(rlnEpochSizeSec: uint) {.async.} =
+      let wakuRlnConfig = WakuRlnConfig(
+        rlnRelayDynamic: false,
+        rlnRelayCredIndex: some(index),
+        rlnRelayUserMessageLimit: 1,
+        rlnEpochSizeSec: rlnEpochSizeSec,
+        rlnRelayTreePath: genTempPath("rln_tree", "waku_rln_relay_4"),
+      )
+
+      let wakuRlnRelay = (await WakuRlnRelay.new(wakuRlnConfig)).valueOr:
+        raiseAssert $error
+
+      let rlnMaxEpochGap = wakuRlnRelay.rlnMaxEpochGap
+      let testProofMetadata = default(ProofMetadata)
+      let testProofMetadataTable = {testProofMetadata.nullifier: testProofMetadata}.toTable()
+
+      for i in 0..<rlnMaxEpochGap:
+        # we add epochs to the nullifierLog
+        let testEpoch =  wakuRlnRelay.calcEpoch(epochTime() + float(rlnEpochSizeSec * i))
+        wakuRlnRelay.nullifierLog[testEpoch] = testProofMetadataTable
+        check: wakuRlnRelay.nullifierLog.len().uint == i + 1
+
+      # clearing it now will be a no-op
+      wakuRlnRelay.clearNullifierLog()
+
+      check: wakuRlnRelay.nullifierLog.len().uint == rlnMaxEpochGap
+
+      # append one more now
```

Comment:
two things:

wondering if all this repeated code can be avoided by reusing the `for i in 0..<rlnMaxEpochGap:`. unsure if removing the `<` may help? afaik the intention here is to add rlnMaxEpochGap + 1 nullifier and see that `clearNullifierLog` doesn't clean them.

and for completeness, perhaps we should check that `clearNullifierLog` does clean the log when expected? meaning i keep adding and `len`()` keeps being `rlnMaxEpochGap`?